state/protocol/badger/snapshot_test.go

```diff
@@ -301,12 +300,68 @@ func TestSealingSegment(t *testing.T) {
 			require.NoError(t, err)
 
 			// sealing segment should be [B2, B3, B4]
-			assert.Len(t, segment, 3)
-			assert.Equal(t, block2.ID(), segment[0].ID())
-			assert.Equal(t, block3.ID(), segment[1].ID())
-			assert.Equal(t, block4.ID(), segment[2].ID())
+			assert.Len(t, segment.Blocks, 3)
+			assert.Equal(t, block2.ID(), segment.Blocks[0].ID())
+			assert.Equal(t, block3.ID(), segment.Blocks[1].ID())
+			assert.Equal(t, block4.ID(), segment.Blocks[2].ID())
+		})
+	})
+
+	// test sealing segment where a block contains a ExecutionReceipt that
+	// references a ExecutionResult contained in a different block.
+	// the sealing segment should contain any ExecutionResults that were
+	// missing from a blocks payload in the SealingSegment.ExecutionResults field
+	// ROOT -> B1(Receipt_A) -> B2[Result_B, Receipt_B, Receipt_A_2] -> B3 -> B4 -> B5 (Seal_B1)
```

Comment:
I'm not sure if this matches the test code. 

This seems to say B2 has several receipts in its payload, but in the test B2 is empty. Maybe we're using different notations here? (I'm assuming `B1(A,B)` means block 1 contains resources A, B)

Reply:
B4 has to be empty but it contains seals and results, I guess you need to update godoc or code.